### PR TITLE
DEPR: Index.is_monotonic for Index.is_monotonic_increasing

### LINKED ISSUE #21335: PeriodIndex.is_monotonic only for monotonic increasing
```
dti = pd.date_range('2016-01-01', periods=3)
pi = dti.to_period('D')

>>> pi.is_monotonic
True
>>> pi[::-1].is_monotonic
False
```

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #21335
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/base.py [hunk 1]
```diff
@@ -1050,17 +1052,27 @@ def is_monotonic(self) -> bool:
...
+
+        Returns
+        -------
+        bool
         """
-        # mypy complains if we alias directly
-        return self.is_monotonic
+        from pandas import Index
+
+        return Index(self).is_monotonic
```
Comment (jbrockmendel) [inline pandas/core/base.py:1075]: should disaptch to is_monotonic_increasing?

File: pandas/core/groupby/ops.py [hunk 2]
```diff
@@ -823,7 +823,7 @@ def groups(self) -> dict[Hashable, np.ndarray]:
     @cache_readonly
     def is_monotonic(self) -> bool:
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py:824]: deprecate here too?

**Comment by jbrockmendel** [comment]
closes #21335 i think

File: pandas/core/base.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/base.py:1075]: Thanks. Fixed.

File: pandas/core/groupby/ops.py (hunk 2 above)
Comment (mroeschke) [reply pandas/core/groupby/ops.py:824]: I'll try to address this one in another pass. This appears to be a private-ish version for groupby specifically.

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**